main.go

```diff
@@ -17,5 +20,10 @@ import (
 //go:generate go run github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs
 
 func main() {
-	plugin.Serve(&plugin.ServeOpts{ProviderFunc: provider.New})
+	err := providerserver.Serve(context.Background(), provider.New, providerserver.ServeOpts{
```

Comment:
Would you mind adding the `-debug` flag here, similarly to https://github.com/hashicorp/terraform-provider-tls/blob/main/main.go#L16 ?

This is not strictly in topic with this PR, but it seems small enough that I thought I'd mention it.

Reply:
Good shout, have added.